modules/bullet/space_bullet.cpp

```diff
@@ -482,7 +482,7 @@ void SpaceBullet::remove_rigid_body(RigidBodyBullet *p_body) {
 
 	int constraints = btBody->getNumConstraintRefs();
 	if (constraints > 0) {
-		WARN_PRINT("A body connected to joints was removed. Ensure bodies are disconnected from joints before removing them.");
```

Comment:
This is still triggered, because removing the rigid body from the physics world (`NOTIFICATION_EXIT_WORLD`) takes place before the joint has a chance to handle removed bodies.

It could be fixed by adding a new signal `_exit_world` to `CollisionObject3D` emitted before the body is removed from physics so the joints can connect to this one instead.

In 2D things are called in the right order already so there's no need for a change.

Reply:
Do you have an MRP where this happens?

The `NOTIFICATION_EXIT_WORLD` is called during `NOTIFICATION_EXIT_TREE`:
https://github.com/godotengine/godot/blob/41e9028868e49669de83c82ba20fd9dcef0d1b8b/scene/3d/node_3d.cpp#L153-L154
But `NOTIFICATION_EXIT_TREE` happens after `tree_exiting()` is emitted:
https://github.com/godotengine/godot/blob/41e9028868e49669de83c82ba20fd9dcef0d1b8b/scene/main/node.cpp#L267-L269
So, with this PR, the `PhysicsBodies` are detached from the `Joint` before they're removed from the space.

The only other scenario I can think of is: changing a `Viewport`'s world. A `NOTIFICATION_EXIT_WORLD` is propagated to the children of the `Viewport`, when a `Viewport`'s world is changed. However, since `Joint`s break when removing and adding them to a `SceneTree` (#41398), I'm not sure it's worth addressing as part of this PR, which is focused on addressing the crashing issue (#44703).

Reply:
Yes, my bad. I think I might have made a mistake when testing on 3.2 and used `tree_exited` instead of `tree_exiting`. I've tested again and it works as you describe.